File: backend/Domain/Entities/Applicant.cs
using backend.Domain.Enums;

namespace backend.Domain.Entities;

public class Applicant
{
    public Guid Id { get; set; }

    public int Age { get; set; }

    public string State { get; set; } = string.Empty;

    public string City { get; set; } = string.Empty;

    public bool IsFirstLicence { get; set; }

    public List<VehicleType> VehicleTypes { get; set; } = [];

    public LicenceJourney? LicenceJourney { get; set; }

    public List<Licence> Licences { get; set; } = [];
}

File: frontend/src/types/applicant.ts
export type VehicleType = 'MCWG' | 'LMV';

export type CreateApplicantRequest = {
  age: number;
  state: string;
  city: string;
  isFirstLicence: boolean;
  vehicleTypes: VehicleType[];
};

export type RuleFailure = {
  rule: string;
  category: string;
  message: string;
};

export type Applicant = {
  id: string;
  age: number;
  state: string;
  city: string;
  isFirstLicence: boolean;
  vehicleTypes: VehicleType[];
  isEligible: boolean;
  failures: RuleFailure[];
  referenceNumber: string;
};